### PR TITLE
Re-label a diagnostic as a period diagnostic

### PR DESCRIPTION
**Pull request by gavinevans**
Related to https://github.com/metoppv/mo-blue-team/issues/80

Description
Add a small CLI and function, primarily for relabelling an instantaneous diagnostic as a period diagnostic. This may be useful when trying to combine diagnostics when a mix of period and instantaneous diagnostics are available.

Testing:
 - [x] Ran tests and they passed OK
 - [x] Added new tests for the new feature(s)

### CODE REVIEW

File: improver/cli/relabel_to_period.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""CLI to relabel a diagnostic as a period diagnostic."""
+
+from improver import cli
+
+
+@cli.clizefy
+@cli.with_output
+def process(cube: cli.inputcube, period: int):
```
Comment (BelligerG) [inline improver/cli/relabel_to_period.py:39]: If you alter this to `process(cube: cli.inputcube, *, period: int)` then `period` should become a kwarg, I think that's make it more consistent with other CLIs in improver and make the app settings in the suite clearer

Comment (gavinevans) [reply improver/cli/relabel_to_period.py]: Thanks @BelligerG. Yes, that does seem to be the standard, so I've amended the interface.

**Review by MoseleyS — CHANGES_REQUESTED** [review]
This CLI does one of four things

Period setting | Instantaneous input | Period input
------- | ---------- | -------
period is specified | Period output | Period replaced on output
period not specified | Instantaneous output | Period output

This is not the behaviour I was expecting. Instead, I thought I'd see something that did this:
Period setting | Instantaneous input | Period input
------- | ---------- | -------
period is specified | Period output | Error: period already set
period not specified | Error: Nothing to do | Error: Nothing to do

I suppose I can't see any point in making the `period` argument optional. If the user doesn't specify one, they shouldn't be calling this CLI at all and have probably made a mistake.

I can see an argument for changing an existing time-bound, but I'd like this to be explicit in the doc-string so we know what behaviour we expect.

**Review by MoseleyS — APPROVED** [review]
Looks good now. All the tests pass too.

**Review by BelligerG — APPROVED** [review]
Looks good to me!

**Pull request merged**